### PR TITLE
Employee qualification controller refactor/and test update

### PR DESCRIPTION
**Pull request by Juanro-Minne**
image
- Tested the accordion that uses the end points all good
- Tested the endpoint via swagger 
- updated unit tests so all are passing
- Eduan told me we can look at the integration tests later as the middleware might be causing some funky issues there.

image
There are two lines of two of the if statements that are yellow preventing max coverage if someone has any suggestions or potential fixes 👍🏻

### CODE REVIEW

File: RR.App/Controllers/HRIS/EmployeeQualificationController.cs [hunk 1]
```diff
@@ -23,107 +24,46 @@ public EmployeeQualificationController(AuthorizeIdentity identity, IEmployeeQual
...
-            return StatusCode(500, ex.Message);
-        }
+        employeeQualificationDto.Id = id;
+        var updatedQualification = await _employeeQualificationService.UpdateEmployeeQualification(employeeQualificationDto);
+        return Ok(updatedQualification);
     }
 
     [Authorize(Policy = "AllRolesPolicy")]
     [HttpGet]
     public async Task<ActionResult<List<EmployeeQualificationDto>>> GetAllEmployeeQualifications()
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/EmployeeQualificationController.cs:45]:
```suggestion
    public async Task<IActionResult> GetAllEmployeeQualifications()
```

File: RR.App/Controllers/HRIS/EmployeeQualificationController.cs [hunk 2]
```diff
@@ -23,107 +24,46 @@ public EmployeeQualificationController(AuthorizeIdentity identity, IEmployeeQual
...
-        catch (Exception ex)
-        {
-            return StatusCode(500, ex.Message);
-        }
+        employeeQualificationDto.Id = id;
+        var updatedQualification = await _employeeQualificationService.UpdateEmployeeQualification(employeeQualificationDto);
+        return Ok(updatedQualification);
     }
 
     [Authorize(Policy = "AllRolesPolicy")]
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/EmployeeQualificationController.cs:43]: This endpoint seems to pull all qualifications and does not filter by anything. It seems like this endpoint is wide open and should probably have a better authorize policy?

Investigate what front-end endpoints are using this call to determine how we can manage the risk of pulling all qualifications.

```suggestion
    [Authorize(Policy = "AdminOrTalentOrJourneyOrSuperAdminPolicy")]
```

File: RR.App/Controllers/HRIS/EmployeeQualificationController.cs [hunk 3]
```diff
@@ -23,107 +24,46 @@ public EmployeeQualificationController(AuthorizeIdentity identity, IEmployeeQual
...
-        {
-            return BadRequest("An error occurred while retrieving qualifications: " + ex.Message);
-        }
+        var qualifications = await _employeeQualificationService.GetAllEmployeeQualifications();
+        return Ok(qualifications);
     }
 
     [Authorize(Policy = "AllRolesPolicy")]
     [HttpGet("{employeeId}")]
     public async Task<ActionResult<List<EmployeeQualificationDto>>> GetEmployeeQualificationByEmployeeId(int employeeId)
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/EmployeeQualificationController.cs:53]:
```suggestion
    public async Task<IActionResult> GetEmployeeQualificationByEmployeeId(int employeeId)
```

File: RR.App/Controllers/HRIS/EmployeeQualificationController.cs (hunk 2 above)
Comment (Juanro-Minne) [reply RR.App/Controllers/HRIS/EmployeeQualificationController.cs:43]: the only instance of it on the frontend is in the actual service its not being used anywhere else currently.

File: RR.App.Tests/Controllers/HRIS/EmployeeQualificationControllerUnitTests.cs [hunk 4]
```diff
@@ -62,30 +50,16 @@ public async Task UpdateEmployeeQualificationReturnsOkObjectResultWithUpdatedQua
...
-
-        var result = await _employeeQualificationController.UpdateEmployeeQualification(1, _employeeQualificationDto);
-
-        var statusCodeResult = Assert.IsType<ObjectResult>(result);
-        Assert.Equal(500, statusCodeResult.StatusCode);
-        Assert.Equal("Test Exception", statusCodeResult.Value);
+        Assert.Equal("An error occurred while updating qualifications", exception.Message);
     }
 
     [Fact(Skip = "Current user needs to be set for validations on endpoint")]
```
Comment (EduanO) [inline RR.App.Tests/Controllers/HRIS/EmployeeQualificationControllerUnitTests.cs:65]:
```suggestion
    [Fact]
```

File: RR.App.Tests/Controllers/HRIS/EmployeeQualificationControllerUnitTests.cs [hunk 5]
```diff
@@ -101,30 +75,6 @@ public async Task GetEmployeeQualificationByEmployeeIdReturnsOkObjectResultWithQ
         Assert.Equal(_employeeQualificationDto, returnValue);
```
Comment (EduanO) [inline RR.App.Tests/Controllers/HRIS/EmployeeQualificationControllerUnitTests.cs:75]:
```suggestion
        Assert.Equivalent(_employeeQualificationDto, returnValue);
```

**Review by EduanO — CHANGES_REQUESTED** [review]
Please check my comments made

**Comment by Juanro-Minne** [comment]
> Please check my comments made
Currently fixing and applying 👍🏻

**Review by madwizzard — APPROVED** [review]

**Review by ReinoRabbit — APPROVED** [review]

**Review by DewanAlberts — APPROVED** [review]

**Review by EduanO — APPROVED** [review]

**Pull request merged**